### PR TITLE
Harmonize mobile button props with desktop

### PR DESCRIPTION
**Pull request by TomTirapani**
+ Deprecate 'modifier' in favour of 'minimal' and 'outlined' props
+ Add support for intents to mobile buttons
+ Refactor implementation and styling for higher consistency with desktop

See issue: https://github.com/xh/hoist-react/issues/2716
See Toolbox PR: https://github.com/xh/toolbox/pull/542

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by amcclain** [comment]
I'm eager to review this as soon as we are ready to merge in changes for our next major release. Thanks!

File: mobile/cmp/button/Button.scss
Comment (amcclain) [inline mobile/cmp/button/Button.scss:132]: Believe we want button-text var here, not bg

File: mobile/cmp/button/Button.scss
Comment (amcclain) [inline mobile/cmp/button/Button.scss:140]: This looked a bit off when in a toolbar, especially when in dark mode, where the darker color appears to be more muted relative to the borders. Looks like outlined desktop buttons w/intents use the same color for text + border

**Review by amcclain — COMMENTED** [review]
Thanks for running through all these changes - great to clean up this old inconsistency.

+ It's a bit of a pain to setup, but a mobile buttons page similar to what we have on desktop would be great to have in terms of quickly scanning through all the button styling variations. It's proven very useful there - especially for combinations like outlined/intent, toolbars, and button groups. Would you mind creating such a page?
+ Wondering if the grid helper buttons (autosize, colChooser) should default to minimal or outlined? Not sure about it, but they always seem a bit heavyweight.

File: mobile/cmp/button/Button.scss
Comment (TomTirapani) [reply mobile/cmp/button/Button.scss:140]: On desktop, the text color is the `-darker` variant of the border's intent color like here - but only for the light theme. I'll adjust mobile to use the `-lighter` variant for both text and border in dark theme, which will get us full parity with desktop.

**Comment by TomTirapani** [comment]
Thanks for taking a look!

Great idea about the buttons page - I've added it to Toolbox, and it was a great help to spot inconsistencies with desktop. I've pushed a number of updates to address issues and the dark theme styling, and think it's ready for another review.

**Comment by TomTirapani** [comment]
Regarding defaulting the helper buttons to `minimal` / `outlined` - I quite like the heavy weight buttons on mobile, where you don't have hover / cursor pointer to communicate interactivity and thus less room for subtlety. But I don't feel too strongly about it, could be convinced otherwise :)

**Comment by amcclain** [comment]
Sorry to have missed you earlier today and to hold this up for one more cycle, but looks like we lost the "pressed" or `:active` styling. IMO that's an important bit of UI feedback and something we want to maintain / improve (similar to your recent updates to menu items, pending updates for grid rows).

It's confusing because of the dual-use of the term "active" for both "depressed to indicate something is turned on" buttons (`--active` suffix) and "currently held down by the user" buttons (`:active` modifier).

I spent a little while tonight seeing if I could make some minimal / obvious changes to get what I was looking for, but decided that I should touch base with you before going any further down the rabbit hole.  Let's sync up tomorrow (Tuesday) AM on it. 

As part of this, I think we'll want/need mobile-specific overrides for `--xh-button-bg-darker/darkest` to go along with the override for `--xh-button-bg` - believe we'll use the "darkest" modifier for `:active` as we do on desktop.

(Also note I pushed a93b7354262f0e00c42cd981c6485790463918fe to add support for a dedicated border-radius bar for buttons and applied it to the mobile buttons)

**Review by amcclain — APPROVED** [review]

**Pull request merged**
